doc/ipythonconsole.rst

```diff
@@ -2,91 +2,159 @@
+#. Copy the kernel's connection file (:file:`{jupyter/runtime/dir/path}/kernel-{pid}.json`) to the machine you're running Spyder on.
 
+   You can get :file:`{jupyter/runtime/dir/path}` by executing ``jupyter --runtime-dir`` in the same Python environment as the kernel.
+   Usually, the connection file you are looking for will be one of the newest in this directory, corresponding to the time you started the external kernel.
 
-.. _console-features:
+   .. image:: images/console/console-connect-remote-step2.gif
+      :alt: Using SCP to copy the connection file to the local machine
 
-==================
-Supported features
-==================
+#. Click :guilabel:`Connect to an existing kernel` from the :guilabel:`Consoles` menu, and browse for or enter the path to the connection file from the previous step.
 
-Any :guilabel:`IPython Console` in Spyder, internally or externally created, supports additional features including:
+   As a convenience, kernel ID numbers (e.g. ``1234``) entered in the connection file path field will be expanded to :file:`{jupyter/runtime/dir/path}/kernal-{id}.json` on your local machine, if you've copied the connection file there.
 
-.. image:: images/console/console-completion.png
-   :alt: Spyder IPython Console, with a popup list of code completion guesses
+   .. image:: images/console/console-connect-remote-step3.gif
+      :alt: Opening the connect to kernel dialog and browsing for the path
 
-* Automatic code completion
-* Real-time function calltips
-* Debugging toolbar integration for launching the debugger and controlling execution flow
+#. Check the :guilabel:`This is a remote kernel (via SSH)` box and type the full hostname you're connecting to (in the form :file:`{username}@{hostname}:{port-number}`).
+   Then, enter *either* :file:`{username}`'s password on the remote machine, or your user SSH keyfile (typically a  :file:`.perm` file); only one is needed to connect.
 
-Spyder-created consoles support even more advanced capabilities, such as:
+   :guilabel:`Port` is the port number on your remote machine that the SSH daemon (``sshd``) is listening on, typically 22 unless you or your administrator has configured it otherwise.
 
-* The :doc:`variableexplorer`, with GUI-based editors for many built-in and third-party Python objects
-* Full GUI integration with the enhanced IPython debugger, ``ipdb``, including viewing and setting normal and conditional breakpoints interactively in any file, a :guilabel:`Breakpoints` pane, and following along with execution flow in the in the :doc:`editor` (see the :doc:`debugging` documentation for more details)
-* The :ref:`User Module Reloader <umr-section>`, which can automatically re-import modified packages and files
-* Inline display of Matplotlib graphics, if the ``Inline`` backend is selected under :menuselection:`Preferences --> IPython console --> Graphics --> Graphics backend`
+   .. image:: images/console/console-connect-remote-step4.gif
```

Comment:
Could you increase the zooming level of this one so the text is readable. The `This is remote kernel` checkbox you're activating is barely readable.

Reply:
I can zoom it a little more, but not by much or it will start to crop out the dialog, and it will also not look as clean as the upsample won't be an integer ratio and it will start to subsample the pixels.

Reply:
Originally I was going to show entering all the connection details (which is what I was trying to demonstrate with my mouse) but I was worried it would be too long so I just used the saved details instead. I agree it doesn't show that much; I mostly made it a GIF to make it consistent with the others. I wasn't too fond of it and was going to redo it actually entering all the details, but I must have forgotten when I was recording it. If you agree, I can redo it showing that, so its more clear what's going on (which also would avoid the need to move the cursor).